java/common.go

```diff
@@ -34,3 +37,29 @@ Outer:
 	}
 	return false
 }
+
+var (
+	// ErrBadHeader is returned by peekHeader when the possible jar is not
+	// actually a zip file.
+	errBadHeader = errors.New("header indicates this file is not a zip")
+	// JAR files are documented as only using the "standard" zip magic number,
+	// so that's all I'm checking for here. There are two other magic numbers
+	// (ending in "\x05\x06" and "\x07\x08" respectively) for zips, but if I
+	// find those in use I'm going to delete this entire package and go flame
+	// some people on IRC.
+	zipHeader = []byte{'P', 'K', 0x03, 0x04}
+)
+
+// PeekHeader looks at the magic prefix of the current file pointed at by the
+// tar.Reader and returns either the entirety of the file as an io.Reader or an
+// error.
+func peekHeader(tr *tar.Reader) (io.Reader, error) {
```

Comment:
What is the point of returning the file as well as the error? Is there another case you're thinking about or is it because we need the contents anyway so might as well return them?